src/plugins/data/server/saved_objects/query.ts

```diff
@@ -9,14 +9,19 @@
 import { ANALYTICS_SAVED_OBJECT_INDEX } from '@kbn/core-saved-objects-server';
 import { SavedObjectsType } from '@kbn/core/server';
 import { savedQueryMigrations } from './migrations/query';
-import { SCHEMA_QUERY_V8_8_0 } from './schemas/query';
+import {
+  SCHEMA_QUERY_V8_8_0,
+  SCHEMA_QUERY_MODEL_VERSION_1,
+  SCHEMA_QUERY_MODEL_VERSION_2,
+} from './schemas/query';
 
 export const querySavedObjectType: SavedObjectsType = {
   name: 'query',
   indexPattern: ANALYTICS_SAVED_OBJECT_INDEX,
   hidden: false,
   namespaceType: 'multiple',
   convertToMultiNamespaceTypeVersion: '8.0.0',
+  switchToModelVersionAt: '8.10.0',
```

Comment:
FWIW this is the default: https://github.com/elastic/kibana/blob/7aff1f788fcf3d1aa2316278392ade590762f3f8/packages/core/saved-objects/core-saved-objects-server-internal/src/apply_type_defaults.ts#L16